### PR TITLE
AFHS-578 added override until date to EligibilityOverride

### PR DESCRIPTION
**Pull request by AmarkumarJagtap**

### CODE REVIEW

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/EligibilityOverrideDTO.java [hunk 1]
```diff
@@ -22,4 +23,8 @@
     @ApiModelProperty(notes = "Overrides the eligibility outcome from eligibility service")
     private EligibilityOutcome eligibilityOutcome;
 
+    @NotNull
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/EligibilityOverrideDTO.java]: Feels like this should also have `@Future` on it too.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityAndEntitlementService.java [hunk 2]
```diff
@@ -91,15 +91,16 @@ public EligibilityAndEntitlementDecision evaluateClaimantForPaymentCycle(Claim c
     }
 
     private CombinedIdentityAndEligibilityResponse getCombinedIdentityAndEligibilityResponse(Claimant claimant,
-                                                                                             EligibilityOverride eligibilityOverride) {
-        if (isOverride(eligibilityOverride)) {
+                                                                                             EligibilityOverride eligibilityOverride,
+                                                                                             LocalDate validDate) {
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityAndEntitlementService.java]: `validDate` isn't a very good name for this variable, its unclear what this variable means in terms of the test, can you rename and/or explain please?

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/EligibilityOverrideTestDataFactory.java [hunk 3]
```diff
@@ -3,17 +3,30 @@
...
                 .eligibilityOutcome(EligibilityOutcome.CONFIRMED)
+                .overrideUntil(OVERRIDE_UNTIL_FIVE_YEARS)
+                .build();
+    }
+
+    public static EligibilityOverride aConfirmedEligibilityWithUntilDate(LocalDate overrideUntil) {
+        return EligibilityOverride.builder()
+                .eligibilityOutcome(EligibilityOutcome.CONFIRMED)
+                .overrideUntil(overrideUntil)
                 .build();
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/EligibilityOverrideTestDataFactory.java]: Is it worth having a default builder method yet which can be reused for all these methods?

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityAndEntitlementService.java (hunk 2 above)
Comment (YetAnotherMatt) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityAndEntitlementService.java]: `eligibleAtDate` - took me a while to find a good name :)

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/model/NewClaimDTOTest.java [hunk 4]
```diff
@@ -66,4 +68,46 @@ void shouldFailToValidateClaimWithNoClaimant() {
...
+        Set<ConstraintViolation<NewClaimDTO>> violations = validator.validate(claim);
+        //Then
+        assertThat(violations).hasTotalViolations(2);
+        assertThat(violations).hasViolation("must not be null", "eligibilityOverride.eligibilityOutcome");
+        assertThat(violations).hasViolation("must not be null", "eligibilityOverride.overrideUntil");
+
+    }
+
+    @Test
+    void shouldFailToValidateClaimWithPastUntilDateOnEligibilityOverride() {
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/model/NewClaimDTOTest.java]: How about `shouldFailToValidateClaimWithEligibilityOverrideUntilDateInPast`?

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/model/NewClaimDTOTest.java [hunk 5]
```diff
@@ -66,4 +68,46 @@ void shouldFailToValidateClaimWithNoClaimant() {
...
+        //Given
+        NewClaimDTO claim = aValidClaimDTOWithEligibilityOverride(
+                EXPECTED_DELIVERY_DATE_IN_TWO_MONTHS,
+                NO_CHILDREN,
+                EligibilityOutcome.CONFIRMED,
+                LocalDate.now());
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/model/NewClaimDTOTest.java]: This date isn't in the past...

**Pull request merged**